subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/AbstractCompleteFileSystemLocationSnapshot.java

```diff
@@ -47,21 +51,72 @@ public CompleteFileSystemLocationSnapshot store(String absolutePath, int offset,
         return this;
     }
 
+    @Override
+    public FileSystemNode asFileSystemNode(String pathToParent) {
```

Comment:
We need to clarify the role of `pathToParent` here: the object already has a path to parent, so what is this parameter do?

Reply:
Why does this method have the same content as `withPathToParent()`?

Reply:
That comes from the current situation that a `CompleteFileSystemLocationSnapshot` is a `FileSystemNode` and a `MetadataSnapshot`. The method `asFileSystemNode` allows converting a snapshot into a file system node, given an path to the parent node in the hierarchy.

`withPathToParent` comes from `FileSystemNode` and changes the path to parent to the new location.